pulsar-client-cpp/python/pulsar.py

```diff
@@ -185,6 +185,10 @@ def __init__(self, dynamicLibPath, authParamsString):
         * `authParamsString`: Comma-separated list of provider-specific
           configuration params
         """
+        if type(dynamicLibPath) is not str:
```

Comment:
Since the code is using the same if logic so many times I was thinking of using something more generic.

```
def checkDataType(var, type, varName):
    if not isinstance(var, type):
        raise ValueError("{0} is expected to be of type {1}".format(varName, str(type))) 


x = " "
checkDataType(x, int, "x")
```

Reply:
Good point, there are a couple of variations, since some argument can also be `None` and that's ok, but that would shorten the code. I'll change it